app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -0,0 +1,217 @@
+import SmartTransactionsController from '@metamask/smart-transactions-controller';
+import {
+  Fee,
+  SmartTransactionStatuses,
+  SmartTransaction,
+} from '@metamask/smart-transactions-controller/dist/types';
+import type { Hex } from '@metamask/utils';
+import {
+  TransactionController,
+  TransactionMeta,
+  TransactionParams,
+} from '@metamask/transaction-controller';
+import log from 'loglevel';
+
+import { decimalToHex } from '../../../../shared/modules/conversion.utils';
+import {
+  SMART_TRANSACTION_CONFIRMATION_TYPES,
+  ORIGIN_METAMASK,
+} from '../../../../shared/constants/app';
+
+export type SubmitSmartTransactionRequest = {
+  transactionMeta: TransactionMeta;
+  smartTransactionsController: SmartTransactionsController;
+  transactionController: TransactionController;
+  isSmartTransaction: boolean;
+  controllerMessenger: any;
+  featureFlags: Record<string, any>;
+};
+
+export async function submitSmartTransactionHook(
+  request: SubmitSmartTransactionRequest,
+) {
+  const {
+    transactionMeta,
+    smartTransactionsController,
+    transactionController,
+    isSmartTransaction,
+    controllerMessenger,
+    featureFlags,
+  } = request;
+  log.info('Smart Transaction - Executing publish hook', transactionMeta);
+  const isDapp = transactionMeta?.origin !== ORIGIN_METAMASK;
+  const { chainId, txParams } = transactionMeta;
+  // Will cause TransactionController to publish to the RPC provider as normal.
+  const useRegularTransactionSubmit = { transactionHash: undefined };
+  if (!isSmartTransaction) {
+    log.info(
+      `Smart Transaction - Skipping hook as it is not a smart transaction on chainId: ${chainId}`,
+    );
+    return useRegularTransactionSubmit;
+  }
+  const { id } = controllerMessenger.call('ApprovalController:startFlow');
+  const smartTransactionStatusApprovalId: string = id;
+  try {
+    const feesResponse = await smartTransactionsController.getFees(
+      { ...txParams, chainId },
+      undefined,
+    );
+    log.info('Smart Transaction - Retrieved fees', feesResponse);
+    const signedTransactions = await createSignedTransactions(
```

Comment:
Can these `createSignedTransactions` be in a `Promise.all` to run concurrently?

Reply:
We could do that, but right now the /getFees endpoint only returns one trade tx fee, since we removed gas fee escalation. Backend also doesn't return any cancelFees at the moment, so ultimately we will be only having one signed transaction here.